### PR TITLE
fix(core): do not add window.cordova on web apps

### PR DESCRIPTION
**Pull request by elylucas**
I wrapped the logic in the legacy handlers to only add the empty Cordova object and the native event handlers to only be executed when it is running on a device. This fixes the issue with Ionic Core wrongly identifying web apps as being identified as Cordova and hybrid apps as brought up in issue #4198.

### CODE REVIEW

File: core/src/tests/legacy.spec.ts [hunk 1]
```diff
@@ -102,8 +102,8 @@ describe('legacy', () => {
 
   it('doc.addEventListener backbutton', done => {
     const AppWeb = class {
-      async addListener(eventName: string) {
-        expect(eventName).toBe('backButton');
+      async addListener(event: any) {
+        expect(event.eventName).toBe('backButton');
```
Comment (dwieeb) [inline core/src/tests/legacy.spec.ts:106]: Why was this change necessary? `addListener()` methods within plugin classes shouldn't have this signature.

Comment (elylucas) [reply core/src/tests/legacy.spec.ts:106]: when I added the androidBridge to the window object (line119), it changed the result that was coming back from the method from just a string to the object.

**Comment by lincolnthree** [comment]
Glad I'm not crazy. I thought I was losing my mind for a while, trying to track down what I'd done that had caused this to start happening.

**Review by dwieeb — APPROVED** [review]

**Comment by dwieeb** [comment]
@lincolnthree Thank you for reporting it!

**Comment by lincolnthree** [comment]
Hey @dwieeb, did this make it into the beta.3 build? Tried it out and it looks like the compiled sources do not include the fix:

image

```
sharktop:topdecked-unified lincoln$ npx cap doctor
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 2.4.6
  @capacitor/core: 2.4.6
  @capacitor/android: 2.4.6
  @capacitor/ios: 2.4.6

Installed Dependencies:

  @capacitor/cli: 3.0.0-beta.3
  @capacitor/core: 3.0.0-beta.3
  @capacitor/android: 3.0.0-beta.3
  @capacitor/ios: 3.0.0-beta.3

[success] iOS looking great! 👌
[success] Android looking great! 👌
sharktop:topdecked-unified lincoln$ 
```

**Comment by lincolnthree** [comment]
Actually hang on. I think there's some weird caching going on here. I see the updated dist/ in node_modules, but the `ionic serve` code does not seem to reflect the change, even with cache disabled in browser.

**Comment by lincolnthree** [comment]
Okay, confirmed! This is working :) Was a weird NPM multi-version precedence thing... Thanks again!
